### PR TITLE
Fetch form in the action form app

### PR DESCRIPTION
**Pull request by paulfalgout**
Shortcut Story ID: [sc-33890][sc-33832]

We do this for patient and preview forms, but were relying on the cache to grab it from the action.

Note: also includes removing the `form.fetch()` in the form service. Which is no longer needed after this PR.

### CODE REVIEW

**Review by shadowhand — APPROVED** [review]

**Comment by paulfalgout** [comment]
this is going to have some breaking tests because they're expecting the stubbed form endpoint and it's not stubbed..

File: test/support/api/forms.js [hunk 1]
```diff
@@ -18,20 +18,20 @@ Cypress.Commands.add('routeForms', (mutator = _.identity) => {
...
     response() {
       return mutator({
         data: getResource(_.find(this.fxTestForms, { id: formId }), 'forms'),
         included: [],
       });
     },
   })
-    .as('routeForm');
+    .as('routeFormByAction');
 });
```
Comment (nmajor25) [inline test/support/api/forms.js:56]: Is this correct?

New `routeFormByAction` command is identical to `routeForm` (now removed), apart from the URL?

**Comment by paulfalgout** [comment]
@nmajor25 errors are legit I think.  Seems like a couple more route fixes

**Comment by nmajor25** [comment]
@paulfalgout weirdly, these failing tests don't fail for me locally. But I just pushed a fix to hopefully the last one 🤞.

**Pull request merged**